### PR TITLE
Added entities, realtional db.

### PR DESCRIPTION
**Pull request by MihaiDruta**

### CODE REVIEW

**Review by nspc97 — APPROVED** [review]

**Review by nspc97 — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -7,7 +7,7 @@
 /.env.*.local
 /public/bundles/
 /var/
-/vendor/
+/vendor/p
```
Comment (SpartakusMd) [inline .gitignore]: You have just comited the folder `vendor` which contains 3rd party modules, and that's 10000 files )) Fix .gitignore file.

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

File: src/Entity/Activity.php [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+{
+    /**
+     * @ORM\Id()
+     * @ORM\GeneratedValue()
+     * @ORM\Column(type="integer")
+     */
+    protected $id;
+
+    /**
+     * @ORM\Column(type="string", length=30)
```
Comment (SpartakusMd) [inline src/Entity/Activity.php]: Usually the names are longer. The default length of `string` is 255. Please remove the length attribute. The change should be reflected in the other classes also.

File: src/Entity/Activity.php [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+     */
+    private $name;
+
+    /**
+     * @ORM\Column(type="string", length=1000)
```
Comment (SpartakusMd) [inline src/Entity/Activity.php]: Type `text` is more appropriate here. Remove the length attribute

File: src/Entity/Activity.php [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $applicationDeadline;
+
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $finalDeadline;
```
Comment (SpartakusMd) [inline src/Entity/Activity.php:43]: What is the difference from `$applicationDeadline`?

File: src/Entity/Activity.php [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+    /**
+     * @ORM\Column(type="string", length=30)
+     */
+    private $status;
```
Comment (SpartakusMd) [inline src/Entity/Activity.php]: What will be the values for this field? For status it's better to have constants as values:

```
public const STATUS_NEW = 0;
public const STATUS_FINISHED = 1;
public const STATUS_CLOSED = 2;

private $status = self::STATUS_NEW;
```

File: src/Entity/Activity.php [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+
+    /**
+     * @ORM\Column(type="integer")
+     */
+    private $idOwner;
```
Comment (SpartakusMd) [inline src/Entity/Activity.php]: `OneToMany` with `User`. Rename the field to `$owner`.

File: src/Entity/Activity.php [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name): self
+    {
+        $this->name = $name;
+
+        return $this;
```
Comment (SpartakusMd) [inline src/Entity/Activity.php]: Not all functions return `$this`. Example: `addActivityTechnology`, `removeActivityTechnology`.

File: src/Entity/Activity.php [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $updatedAt;
+
+    /**
+     * @var \Doctrine\Common\Collections\Collection|Technology[]
+     * @ORM\ManyToMany(targetEntity="Technology")
+     */
+    protected $activityTechnologies;
```
Comment (SpartakusMd) [inline src/Entity/Activity.php]: Why not `$technologies`?

File: src/Entity/Activity.php [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+
+    /**
+     * @var \Doctrine\Common\Collections\Collection|Type[]
+     * @ORM\ManyToMany(targetEntity="Type")
+     */
+    protected $activityTypes;
```
Comment (SpartakusMd) [inline src/Entity/Activity.php]: Why not `$types`?

File: src/Entity/Activity.php [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+
+    public function __construct()
+    {
+        $this->activityTechnologies = new ArrayCollection();
+        $this->activityTypes = new ArrayCollection();
```
Comment (SpartakusMd) [inline src/Entity/Activity.php]: Fields `createdAt` and `updatedAt` should be initialized in the constructor.

File: src/Entity/User.php [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+
+    /**
+     * @ORM\Column(type="string", length=30)
+     */
+    private $role;
+
+    /**
+     * @ORM\Column(type="string", length=30)
+     */
+    private $seniority;
```
Comment (SpartakusMd) [inline src/Entity/User.php]: Use constants.

File: src/Entity/User.php [hunk 12]
```diff
@@ -0,0 +1,194 @@
...
+     */
+    private $createdAt;
+
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $updatedAt;
+
+    /**
+     * @var \Doctrine\Common\Collections\Collection|Technology[]
```
Comment (SpartakusMd) [inline src/Entity/User.php]: Import `\Doctrine\Common\Collections\Collection` in the `use` block so here you will have `Collection|Technology[]`

File: src/Migrations/Version20190430070122.php [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf(
+            $this->connection->getDatabasePlatform()->getName() !== 'mysql',
+            'Migration can only be executed safely on \'mysql\'.'
+        );
+
+        $this->addSql('CREATE TABLE technology_user (technology_id INT NOT NULL, user_id INT NOT NULL, INDEX IDX_2B90AC434235D463 (technology_id), INDEX IDX_2B90AC43A76ED395 (user_id), PRIMARY KEY(technology_id, user_id)) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci ENGINE = InnoDB');
+        $this->addSql('ALTER TABLE technology_user ADD CONSTRAINT FK_2B90AC434235D463 FOREIGN KEY (technology_id) REFERENCES technology (id) ON DELETE CASCADE');
+        $this->addSql('ALTER TABLE technology_user ADD CONSTRAINT FK_2B90AC43A76ED395 FOREIGN KEY (user_id) REFERENCES user (id) ON DELETE CASCADE');
+        $this->addSql('DROP TABLE activity');
```
Comment (SpartakusMd) [inline src/Migrations/Version20190430070122.php]: It seems when you first ran the migrations you had tables in database. Cleanup them:
- Remove the migrations.
- Remove everything from the database.
- Generate the migrations.
- Commit.

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

File: src/Entity/Activity.php (hunk 5 above)
Comment (MihaiDruta) [reply src/Entity/Activity.php]: @SpartakusMd 
What type should we use for fields with constants? integer?

File: src/Entity/Activity.php (hunk 4 above)
Comment (MihaiDruta) [reply src/Entity/Activity.php:43]: @SpartakusMd 
applicationDeadline - The deadline for job application
finalDeadline - The deadline for the job
We took it from Pentabee Project Info document

File: src/Entity/Activity.php (hunk 8 above)
Comment (MihaiDruta) [reply src/Entity/Activity.php]: @SpartakusMd 
In that case we would have a field named `$technologies` both in user and activity tables. We used `$activityTechnologies` and  `$userTechnologies` instead for a better understanding.

File: src/Migrations/Version20190430070122.php (hunk 13 above)
Comment (SpartakusMd) [reply src/Migrations/Version20190430070122.php]: Don't forget about migrations.

File: src/Entity/Activity.php [hunk 14]
```diff
@@ -48,7 +48,6 @@ class Activity
     private $status = self::STATUS_NEW;
 
     /**
-     * @ORM\Column(type="integer")
```
Comment (SpartakusMd) [inline src/Entity/Activity.php]: Please commit the last migration.

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

File: src/Entity/Activity.php (hunk 14 above)
Comment (MihaiDruta) [reply src/Entity/Activity.php]: Done.

**Review by SpartakusMd — APPROVED** [review]

**Pull request merged**